packages/nimble-components/src/table-column/base/types.ts

```diff
@@ -24,6 +24,7 @@ export interface TableColumnWithPlaceholderColumnConfig {
  */
 export interface DelegatedEventEventDetails {
     originalEvent: Event;
+    rowRecordId?: string;
```

Comment:
What is the scenario this is undefined for users?
A rendered row always has row ids right? Either an index created by the table or the row ids defined by the user? @mollykreis 

Seems like this should not be nullable. Maybe we should not fire an event in the scenario that somehow rowId is undefined?

Reply:
If `rowRecordId` should always be defined, then maybe we should throw an error if we find an undefined value. That seems better than silently skipping the event.

Reply:
Brainstormed with Jesse a bit and I think we have a direction to go:

- Looking at the cell-view as a standalone component, it is completely valid for the component to have a recordId that is undefined. From a unit testing perspective we should have tests for the recordId defined and undefined showing it's behavior (fires and doesn't fire an event respectively).
-  We should have integration level jasmine tests for this feature testing the different scenarios for recordId, i.e. the data including recordId and the data not including recordId (and falling back to table generated record ids).
- With the above then we don't throw an exception here, each component in isolation and in integration is behaving as defined and tested.

> we should throw an error if we find an undefined value

This isn't hard and fast rule we have defined. For example all the validity infrastructure on tables avoids throwing exceptions as there can be a settling phase for property setting. Also on the next line if column is undefined the code is silently ignoring firing the error.